app/models/user.rb

```diff
@@ -37,11 +37,18 @@ def get_coords
   end
 
   def set_profile_picture
-    image = ImageService.new.user_image
-    if !image[:error]
-      self.profile_picture = image[:data][:attributes][:raw_image]
+    begin
+      image = ImageService.new.user_image
+    rescue Faraday::ConnectionFailed
+      RetryImageServiceJob.perform_in(15.seconds, self.id)
     else
-      self.profile_picture = nil
+      if self.profile_picture
+        return
+      elsif !image[:error]
+        self.update(profile_picture: image[:data][:attributes][:raw_image])
```

Comment:
If no error in retrieving it then update the user's profile picture to the image. Nice!